en/setup/vscode.md

```diff
@@ -0,0 +1,55 @@
+# Visual Studio Code IDE
+
+Visual Studio Code (VSCode) is a powerful cross-platform source code editor/IDE that can be used for PX4 development on Linux and macOS (Windows support coming soon).
+
+PX4 has been set up with all the definition files that you need to install needed VSCode extensions, and and start developing.
+This topic explains how to get started.
+
+<!-- 
+- why VSCode vs Qt creator? 
+- Do we recommend VSCode?
+- what had to happen to make this setup work - ie presumably the directory contains definition files that vscode needs to know to set everything up?
+-->
+
+
+## Preconditions
+
+You must already have installed the command line [PX4 developer environment](../setup/dev_env.md) for your platform and downloaded the *Firmware* source code repo.
+
+<!-- xcode? -->
+
+## Installation & Setup
+
+1. [Download and install VSCode](https://code.visualstudio.com/) (you will be offered the correct version for your OS).
+1. Add the PX4 source code into VSCode. 
```

Comment:
Let's keep it simple and only recommend opening the folder. The workspace allows you to have a collection of projects and I don't think we need to go into it for the vast majority.

Reply:
Agreed. Fixed